src/HealthGPS/person.cpp

```diff
@@ -56,14 +60,29 @@ double Person::get_risk_factor_value(const core::Identifier &key) const {
     throw std::out_of_range("Risk factor not found: " + key.to_string());
 }
 
-float Person::gender_to_value() const noexcept {
+float Person::gender_to_value() const {
+    if (gender == core::Gender::unknown) {
+        throw core::HgpsException("Gender is unknown.");
+    }
     return gender == core::Gender::male ? 1.0f : 0.0f;
 }
 
-std::string Person::gender_to_string() const noexcept {
+std::string Person::gender_to_string() const {
+    if (gender == core::Gender::unknown) {
+        throw core::HgpsException("Gender is unknown.");
+    }
     return gender == core::Gender::male ? "male" : "female";
 }
 
+float Person::sector_to_value() const {
+    if (sector == core::Sector::unknown) {
+        throw core::HgpsException("Sector is unknown.");
+    }
+    return sector == core::Sector::urban ? 0.0f : 1.0f;
```

Comment:
Given that it's a bit ambiguous as to what 0 and 1 mean in this context, do we maybe want to rename the sector property to "is_urban" or something?

I suppose the downside of that is that it wouldn't be particularly extensible if more sectors are added in future.

Reply:
Exactly that. Also the choice of rural = 1 is completely arbitrary, and based on how the coefficient data inputs were computed. Worth revisiting, but not now.